tools/list_apis.py

```diff
@@ -0,0 +1,520 @@
+import argparse
+import inspect
+import importlib
+import sys
+
+from types import ModuleType
+from collections import namedtuple
+from typing import List, Any, Set
+
+
+CrawlError = namedtuple("CrawlError", ("reason", "path"))
+CrawlResult = namedtuple("CrawlResult", ("public", "private", "errors", "module"))
+
+
+def err_print(*args: Any) -> None:
+    print(*args, file=sys.stderr)
+
+
+system_module_names = {
```

Comment:
Is there any better way other than listing out all system modules? for example using `modules` dunder methods

Reply:
I couldn't [find any](https://stackoverflow.com/questions/6463918/how-can-i-get-a-list-of-all-the-python-standard-library-modules), `isort` (the import ordering checker) does something similar: https://github.com/PyCQA/isort/tree/main/isort/stdlibs

What do you mean `modules` dunder methods?

Reply:
I haven't dug deeper but out of top of my head we can check if `module.__file__` is coming from one of the site-packages other than of `site-package/torch`.

Or even simpler can we check if the module is coming from one of the torch file? Ultimately we are only interested in the torch module right?

Reply:
In case it helps, here is the version of this script that numpy uses: https://github.com/numpy/numpy/blob/main/numpy/tests/test_public_api.py